packages/dds/tree/src/core/tree/detachedFieldIndex.ts

```diff
@@ -157,15 +248,54 @@ export class DetachedFieldIndex {
 					) === undefined,
 					0x7ce /* Detached node ID already exists in index */,
 				);
-				setInNestedMap(this.detachedNodeToField, nodeId.major, nodeId.minor + i, root + i);
+				setInNestedMap(this.detachedNodeToField, nodeId.major, nodeId.minor + i, {
+					root: root + i,
+					latestRelevantRevision: revision,
+				});
+				this.updateLatestRevision(
+					{ major: nodeId.major, minor: nodeId.minor + i },
+					revision,
+				);
 			}
 		}
 		return root;
 	}
 
+	/**
+	 * Updates the latest revision that is relevant to the provided root
+	 */
+	public updateLatestRevision(id: Delta.DetachedNodeId, revision: RevisionTag | undefined): void {
+		const fieldEntry =
+			tryGetFromNestedMap(this.detachedNodeToField, id.major, id.minor) ??
+			fail("detached node id does not exist in the detached field index");
```

Comment:
Please try to use `assert` statements instead of `fail`. Assert messages get minified by our tooling so they don't cause as much package size bloat.